File: tools/liked_posts.py
#!/usr/bin/env python3
"""Fetch the posts an account has liked. Prints one indexed line per post —
see post_index.print_indexed for the exact format and how to reference a post.

Bluesky only serves a repo's likes to that repo's owner, so this works for the
logged-in account only; another handle comes back as an HTTP error.

    python3 liked_posts.py alice.bsky.social --limit 20
"""
import argparse
import sys

from bsky import BskyError, get
from post_index import print_indexed


def get_actor_likes(actor: str, limit: int) -> list[dict]:
    data = get("app.bsky.feed.getActorLikes", {"actor": actor, "limit": limit})
    # Same feed envelope as getAuthorFeed: [{post, reason?}, ...]
    return [item["post"] for item in data.get("feed", []) if item.get("post")]


def main() -> None:
    parser = argparse.ArgumentParser(description=__doc__)
    parser.add_argument("actor")
    parser.add_argument("--limit", type=int, default=10)
    args = parser.parse_args()

    print_indexed(get_actor_likes(args.actor, args.limit))


if __name__ == "__main__":
    try:
        main()
    except BskyError as e:
        print(f"likes fetch failed: {e}", file=sys.stderr)
        sys.exit(1)

File: agent/.opencode/tools/liked-posts.ts
import { tool } from "@opencode-ai/plugin"
import { $ } from "bun"

// Thin wrapper around tools/liked_posts.py — located via the git worktree.
// Bluesky serves a repo's likes to its owner only, so this works for the
// logged-in account; any other handle comes back as an error from the Python side.
export default tool({
  description:
    "Fetch the posts an account has liked — the inverse of liked-by, which " +
    "returns the accounts that liked one post. Only works for the logged-in " +
    "account. Returns one line per post: `[N] @handle (K likes) at://uri — " +
    "preview`. To show a post in your reply, write just its bracketed index " +
    "like [N]; the UI renders it as a live card.",
  args: {
    actor: tool.schema.string().describe("Handle, e.g. 'alice.bsky.social'"),
    limit: tool.schema.number().describe("Max posts to return").default(10),
  },
  async execute(args, context) {
    const script = `${context.worktree}/tools/liked_posts.py`
    // Guard limit — the model sometimes omits it and a schema default doesn't
    // always reach here, which would send the literal "undefined" (argparse
    // exit 2).
    const limit = Number.isFinite(args.limit) ? Math.trunc(args.limit as number) : 10
    return await $`python3 ${script} ${args.actor} --limit ${String(limit)}`.env({ ...process.env, ACTING_SESSION: context.sessionID }).text()
  },
})
